### ISSUE TITLE
`# %% [markdown]` is not recognized

### ISSUE THREAD

**Issue by matrix-ranger** [nteract/hydrogen#2097]
### Description

`# %% [markdown]` does not display markdown results in code scripts. `# %% markdown` displays `%% markdown` at the beginning of the cell which is not desired. Only `# %% markdown` following an empty line works well.

<img width="331" alt="image" src="https://user-images.githubusercontent.com/48684632/117646434-543b3100-b1be-11eb-9ccf-5bf26da1d152.png">
<img width="393" alt="image" src="https://user-images.githubusercontent.com/48684632/117646481-64531080-b1be-11eb-892f-42b37fc316bc.png">
<img width="291" alt="image" src="https://user-images.githubusercontent.com/48684632/117646516-703ed280-b1be-11eb-9c32-2388495f56b0.png">

This is quite annoying because I am exporting .ipynb to .R using Jupytext, which uses `# %% [markdown]` as the indicator of a markdown cell.

### Versions

macOS 11.3.1
Atom and Hydrogen the lastest

**Comment by aminya** [comment]
Why don't you import ipynb directly inside Atom?

**Comment by matrix-ranger** [comment]
I mean it is a feature explicitly written in the documents of Hydrogen. It does not work as said in the documents. Then it is a bug. It does not matter why I did not import ipynb directly inside Atom (the reason is jupytext can sync between ipynb and py files). And I needn't upload a file. You can just reproduce by typing `# %% [markdown]` and write something in the next line in a .py file.

**Comment by aminya** [comment]
> And I needn't upload a file. You can just reproduce by typing # %% [markdown] and write something in the next line in a .py file.

I cannot fix a bug with a reproduction. After uploading the file, please explain what you want to happen with markdown cells? For example, should it render the markdown?

**Comment by matrix-ranger** [comment]
Yes. It should render the markdown and show preview below the cell. It works for `# %% markdown` but not for `# %% [markdown]`. Yet both are recommended delimiters according to the doc.
[a.md](https://github.com/nteract/hydrogen/files/6557766/a.md)
Sorry. I tried to upload the `a.py` file but Github says they don't support that format in the comments. So I renamed it to `a.md`. Please rename it back to `a.py` to reproduce the issue.
<img width="379" alt="Fig2021-05-28_10-25-33" src="https://user-images.githubusercontent.com/48684632/119920907-ab534b00-bf9f-11eb-927f-4afc363a30ff.png">
If you click ctrl/cmd+enter on Line 5, the markdown preview appears below. However, if you click ctrl/cmd+enter on Line 2, only a tick sign appears without the preview.
